### PR TITLE
Correct incorrect assertion in VecDeque::wrap_copy

### PR DESCRIPTION
**Pull request by mystor**
I think this should fix the test failures in debug mode from #29492

The assertion was written incorrectly, and I don't like the way the new assertion is written, but I _think_ it does the right thing now.

### CODE REVIEW

**Comment by bluss** [comment]
Thanks!

@bors r+

File: src/libcollections/vec_deque.rs [hunk 1]
```diff
@@ -175,9 +175,11 @@ impl<T> VecDeque<T> {
     /// (abs(dst - src) + len) must be no larger than cap() (There must be at
     /// most one continuous overlapping region between src and dest).
     unsafe fn wrap_copy(&self, dst: usize, src: usize, len: usize) {
-        debug_assert!(
-            (if src <= dst { dst - src } else { src - dst }) + len <= self.cap(),
-            "dst={} src={} len={} cap={}", dst, src, len, self.cap());
+        #[allow(dead_code)]
+        fn diff(a: usize, b: usize) -> usize {if a <= b {b - a} else {a - b}}
```
Comment (nagisa) [inline src/libcollections/vec_deque.rs]: not sure its worth very much making this function instead of keeping it inlined.

Comment (mystor) [inline src/libcollections/vec_deque.rs]: I agree. Not a big fan of this code, but it works, which is better than the old code :)

Comment (bluss) [inline src/libcollections/vec_deque.rs]: It's called twice, so I think it's ok.

File: src/libcollections/vec_deque.rs [hunk 2]
```diff
@@ -175,9 +175,11 @@ impl<T> VecDeque<T> {
     /// (abs(dst - src) + len) must be no larger than cap() (There must be at
     /// most one continuous overlapping region between src and dest).
     unsafe fn wrap_copy(&self, dst: usize, src: usize, len: usize) {
-        debug_assert!(
-            (if src <= dst { dst - src } else { src - dst }) + len <= self.cap(),
-            "dst={} src={} len={} cap={}", dst, src, len, self.cap());
+        #[allow(dead_code)]
```
Comment (mystor) [inline src/libcollections/vec_deque.rs]: I'm not sure if this is necessary come to think of it...

**Pull request merged**